src/widgets/block.rs

```diff
@@ -143,8 +161,33 @@ impl<'a> Block<'a> {
         if self.borders.intersects(Borders::BOTTOM) {
             inner.height = inner.height.saturating_sub(1);
         }
+
+        inner.x = inner.x.saturating_add(self.padding.horizontal);
+        inner.y = inner.y.saturating_add(self.padding.vertical);
+
+        inner.width = inner.width.saturating_sub(self.padding.horizontal * 2);
+        inner.height = inner.height.saturating_sub(self.padding.vertical * 2);
+
         inner
     }
+
+    pub fn padding(mut self, padding: u16) -> Block<'a> {
```

Comment:
What if we condensed these 3 functions to one function:

```
fn padding(mut self, padding: Padding)
```

And added some constructors for padding:
```
impl Padding {
   pub fn new(x:u16, y: u16) -> Self {...}
   pub fn horizontal(val: u16) -> Self {...}
   pub fn vertical(val: u16) -> Self {...}
   pub fn uniform(val: u16) -> Self {...}
}
```
This would simplify the api for block, and also allow for easier future expansion of the concept (e.g. left, right, top, bottom being independent values).

Reply:
Good point 👍🏼

Reply:
Implemented in 981f1e4bf40e59e685249d226fc18b550621d973

Reply:
@orhun - i think you need to resolve this conversation, I don't have buttons for it :D